token-swap/program/src/processor.rs

```diff
@@ -2057,21 +2096,27 @@ mod tests {
                 host_fee_numerator,
                 host_fee_denominator,
             };
+            let curve = ConstantProductCurve {};
             let swap_curve = SwapCurve {
                 curve_type: CurveType::ConstantProduct,
-                calculator: Box::new(curve.clone()),
+                calculator: Box::new(curve),
             };
-            curve.trade_fee_numerator = trade_fee_numerator;
             let owner_key = &user_key.to_string();
-            let valid_constant_product_curves = &[curve];
+            let valid_curve_types = &[CurveType::ConstantProduct];
             let constraints = Some(FeeConstraints {
                 owner_key,
-                valid_constant_product_curves,
-                valid_flat_curves: &[],
-                valid_stable_curves: &[],
+                valid_curve_types,
+                fees: &fees,
             });
-            let mut accounts =
-                SwapAccountInfo::new(&user_key, swap_curve, token_a_amount, token_b_amount);
+            let mut bad_fees = fees.clone();
+            bad_fees.trade_fee_numerator = trade_fee_numerator;
```

Comment:
This is a bit confusing. How about instead of decrementing the trade fee numerator for the fee constraints, you deprecate it for the bad fees object?